lib/util/gcp/storage/client/client.go

```diff
@@ -0,0 +1,41 @@
+package client
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+)
+
+type transport struct {
+	URL *url.URL
+}
+
+func (t transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.URL.Host = t.URL.Host
+	req.URL.Scheme = t.URL.Scheme
+	return http.DefaultTransport.RoundTrip(req)
+}
+
+// New creates a storage.Client instance and, if the GCP_GCS_EMULATOR_URL env
+// var is set, then a custom HTTPClient is used, which forwards all requests to
+// the provided URL.
+// Note: [this bug](https://github.com/googleapis/google-cloud-go/issues/2476)
+// prevents the `STORAGE_EMULATOR_HOST` environment variable from working
+// correctly, so this alternative approach is needed when using a GCS emulator.
+func New(ctx context.Context) (*storage.Client, error) {
+	var opts []option.ClientOption
+	if rawURL := os.Getenv("GCP_GCS_EMULATOR_URL"); len(rawURL) > 0 {
```

Comment:
I think hiding this behind an undocumented env var is better than exposing it as a config param. I haven't managed to find any other reasonable way to work around this bug (and I'm glad this works).

Reply:
I can live with it, sounds like some grossness we need to deal with for now. Might be worth opening an issue once this is merged to revisit this later.